src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -233,10 +286,85 @@ public static Text of(Object... objects) {
             }
         }
 
+        if (builder.children.size() == 1) {
+            // Single content, reduce Text depth
+            return builder.children.get(0);
+        }
         return builder.build();
     }
 
     /**
+     * Builds a {@link Text} from a given array of content objects. This method
+     * assumes, that all elements of the array are content and not formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param objects The array that contains the content
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Object... objects) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : objects) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Builds a {@link Text} from a given {@link Iterable}. This method
+     * assumes, that all elements of the iterable are content and not
+     * formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param contents The iterable that contains the content
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Iterable<?> contents) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : contents) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well. Useful for lazy fetching replacements.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Function<String, ?> replacements) {
```

Comment:
> We're not on Java 8 yet, that's not a valid concern. For all we know, the Sponge java 8 version is a few weeks away, and if your PR requires java 8 it won't be merged until then.

I don't use any java 8 features yet, but since there a active works on implementing that I thought I could add some basic support for this.

> I am all for adding power to the Text API, but not in places where it is already possible to do what people want. An attitude of "why should I prevent it" instead of "why should I add it", in my mind, leads to cluttered and unclear code.

> I don't understand why you "need" the function in TextArgument.apply, you could just as well have a simple Map<String, ?> and look up items from it, as I suggested in my first comment.

Of course I could do it using a map as well. A map is basically the same as a function, but a little bit more limited.
Sorry, but I really don't see a difference in complexity/readbility in function.apply(x) and map.get(x).

=> No Cons

Pros:
Function offers only one method, the only one you need at that point. There is no need to get the size of the map, the keys, the entries or anything else except the pure data you asked from the map. Less possible methods => less complexity.

As mentioned above, the possibility to do expensive stuff, key redirecting, value caching and especially doing key based replacements are pros. https://github.com/SpongePowered/SpongeAPI/pull/808/files#r38813818
Using a function can be very helpful/necessary for (de-)serialization of customizable Placeholder/TextArguments.

> it is already possible to do what people want

The example (link, bottom part) isn't possible without that changes.

> An attitude of "why should I prevent it" instead of "why should I add it", in my mind, leads to cluttered and unclear code.

We both have different opinions on the con, but there are pros as well

However, if there is a common opinion that my reasons are not valid, then we can switch over to maps of course.
I'm not perfect after all. IMO Functions are just less restricting.